zebra-chain/src/parameters/network/testnet.rs

```diff
@@ -139,6 +144,20 @@ impl ParametersBuilder {
         self
     }
 
+    /// Sets the network name to be used in the [`Parameters`] being built.
+    pub fn with_network_magic(mut self, network_magic: Magic) -> Self {
+        assert!(
+            [magics::MAINNET, magics::TESTNET, magics::REGTEST]
```

Comment:
The testnet magic should be allowed (it's the default value anyways if this method isn't called before `.finish()`), but maybe only on compatible testnets? (currently where the activation heights are the same as the default testnet, as we add more fields, the target difficulty, equihash parameters, and other parameters may also be relevant).

```suggestion
            [magics::MAINNET, magics::REGTEST]
```